dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/AtomicityPropagation.java

```diff
@@ -73,6 +71,7 @@ public Event visitMemEvent(MemEvent e) {
 			// Accesses to an atomic object without a concrete mo are considered SC
     		if(atomics.contains(e.getAddress()) && e.canRace()) {
     			e.addFilters(Tag.C11.ATOMIC, Tag.C11.MO_SC);
+    			e.removeFilters(Tag.C11.NONATOMIC);
     		}
```

Comment:
I don't think this is enough. We have this special `mo` member in `MemEvent` which should get changed. Otherwise `e.canRace` will continue to return `true`.